services/beaconchain/src/main/java/tech/pegasys/teku/services/beaconchain/BeaconChainController.java

```diff
@@ -642,6 +643,13 @@ protected void initAttesterSlashingPool() {
                 .reversed());
     blockImporter.subscribeToVerifiedBlockAttesterSlashings(attesterSlashingPool::removeAll);
     attesterSlashingPool.subscribeOperationAdded(forkChoice::onAttesterSlashing);
+    if (beaconConfig.validatorConfig().isShutdownWhenValidatorSlashedEnabled()) {
```

Comment:
Maybe make some method like `initSlashingProtection()`, put both subscriptions in it, and add it to run list somewhere in the end after `initActiveValidatorTracker` for example, as pools are global vars anyway. 
There is already some mess here and there but I think this way it will look better.

Reply:
Makes sense. Done